Convex_hull_3/include/CGAL/convex_hull_3.h

```diff
@@ -857,45 +955,67 @@ template <class InputIterator, class Polyhedron_3, class Traits>
 void convex_hull_3(InputIterator first, InputIterator beyond,
                    Polyhedron_3& polyhedron,  const Traits& traits)
 {
-  typedef typename Traits::Point_3                Point_3;  
+  typedef typename Traits::Point_3                Point_3;
   typedef std::list<Point_3>                      Point_3_list;
   typedef typename Point_3_list::iterator         P3_iterator;
+  typedef std::pair<P3_iterator,P3_iterator>      P3_iterator_pair;
+
 
   Point_3_list points(first, beyond);
-  CGAL_ch_precondition(points.size() > 3);
+  clear(polyhedron);
 
-  // at least 4 points 
-  typename Traits::Collinear_3 collinear = traits.collinear_3_object();
-  typename Traits::Equal_3 equal = traits.equal_3_object();
+  if(points.size() == 0)
+    return;
+  else if(points.size() == 1 || points.size() == 2)
+    internal::Convex_hull_3::add_isolated_points(points, polyhedron);
```

Comment:
what if the two points are identical?

Reply:
actually I do not see why you changed the logic of the initial code. what if I have 100 identical points?